newsletter/2020-03.md

```diff
@@ -0,0 +1,165 @@
+
+We are aiming to approve AIP-162 at the end of March. If you have feedback,
+please [leave a comment]().
+
+[aip-162]: ../aip/0162.md
+
+### AIP-163: Validating changes
+
+> Occasionally, a user wants to validate an intended change to see what the
+> result will be before actually making the change. For example, a request to
+> provision new servers in a fleet will have an impact on the overall fleet
+> size and cost, and could potentially have unexpected downstream effects.
+
+[Validating changes][aip-163] is the concept of performing a "dry run" on an
+API request, and returning back the result without making any changes.
+
+This is effectively a migration of long-standing guidance that was not very
+fleshed out. The previous API design guide mentioned `validate_only` as an
+entry in the standard fields table with minimal explanation. This proposal
+fleshes that out with an example, and addresses some questions we have received
+over time.
+
+**Summary:** [AIP-163][] offers guidance for "dry run" requests using a
+`bool validate_only` field in the request message, and addresses some common
+situations around permissions and incomplete output.
+
+We are aiming to approve AIP-163 at the end of March. If you have feedback,
+please [leave a comment]().
+
+[aip-163]: ../aip/0163.md
+
+### AIP-165: Criteria-based delete
+
+> Occasionally, an API may need to provide a mechanism to delete a large number
+> of resources based on some set of filter parameters, rather than requiring
+> the individual resource name of the resources to be deleted.
+
+[Criteria-based delete][aip-165] is designed to address problems where a user
+needs to delete so many items that [batch delete][aip-235] simply becomes
+infeasible.
+
+This proposal represents a concession of sorts: we have long tried to avoid
+delete-by-filter because of concerns that users would accidentally delete
+important records and be unable to recover them. However, several APIs have had
+legitimate reasons to need to do this, because the number of records the user
+would want to delete was so high.
+
+**Summary:** [AIP-165][] offers guidance for criteria-based delete, with a
+common name (`Purge`) and structure, and mandates a dry-run by default with a
+`force` flag to actually perform the purge.
+
+We are aiming to approve AIP-165 at the end of March. If you have feedback,
+please [leave a comment]().
+
+[aip-165]: ../aip/0165.md
+[aip-235]: ../aip/0235.md
+
+### AIP-194: Retries for gRPC clients
+
+> RPCs sometimes fail. When one does, the client performing the RPC needs to
+> know whether it is safe to retry the operation. When status codes are used
+> consistently across multiple APIs, clients can respond to failures
+> appropriately.
+
+[Retries for gRPC clients][aip-194] handles how to deal with errors, and in
+particular when a client can transparently retry those errors.
+
+This proposal came about because of a need for first-party client libraries to
```

Comment:
Possibly "Google" rather than "first-party"?

Reply:
Nah. Applies to anyone using gRPC.

Reply:
Ah, I thought you'd specifically been meaning "Google" by "first-party". Presumably third-party client libraries could make use of it as well, just with less confidence of getting the right settings.

Reply:
Right. The guidance is general. I suppose I could just remove the word "first-party". (And honestly, since this is Google, I could actually say Google, now that I think about it more.)

Reply:
I ended up adjusting this language slightly.